.github/workflows/release-branch-tests.yml

```diff
@@ -0,0 +1,156 @@
+          # Swap single and double quotes because bash doesnt interpret correctly
+          full_branch_list="${full_branch_list//\'/__SINGLE_QUOTE__}"
+          full_branch_list="${full_branch_list//\"/\'}"
+          full_branch_list="${full_branch_list//__SINGLE_QUOTE__/\"}"
+          # now convert it to a bash array and add the main branch
+          full_branch_list=($(echo "$full_branch_list" | jq -r '.[]'))
+          for item in "${full_branch_list[@]}"; do   echo "$item"; done
+          full_branch_list+=("main")
+          # Convert the array to a string
+          full_branch_list_string=$(printf "'%s', " "${full_branch_list[@]}")
+          full_branch_list_string="[${full_branch_list_string%, }]"
+          # Assign the string to a variable
+          echo $full_branch_list_string
+          echo "full_branch_list=$full_branch_list_string" >> $GITHUB_OUTPUT
+
+      - name: "[ANNOTATION] ${{ github.event.repository.name }} - branches to test"
+        run: |
+          title="${{ github.event.repository.name }} - branches to test"
+          message="The workflow will run tests in ${{ inputs.workflows_to_run }} for the following branches of the ${{ github.event.repository.name }} repo: ${{ steps.add-main.outputs.full_branch_list }}"
+          echo "::notice $title::$message"
+
+  kick-off-ci:
+    needs: [fetch-latest-branches]
+    name: Kick-off CI
+    runs-on: ubuntu-latest
+
+    strategy:
+      # must run CI 1 branch at a time b/c the workflow-dispatch polls for
+      # latest run for results and it gets confused when we kick off multiple runs
+      # at once. There is a race condition so we will just run in sequential order.
+      max-parallel: 1
+      fail-fast: false
+      matrix:
+        workflow_name: ${{ fromJson(inputs.workflows_to_run) }}
+        branch: ${{ fromJSON(needs.fetch-latest-branches.outputs.branches_to_test) }}
+
+    steps:
+      - name: "[DEBUG] Print Inputs"
+        run: |
+          echo "matrix.branch:           ${{ matrix.branch }}"
+          echo "matrix.workflow_name:    ${{ matrix.workflow_name }}"
+
+      - name: Check out ${{ matrix.branch }}
+        uses: actions/checkout@v3
+        with:
+          ref: ${{ matrix.branch }}
+
+      # We have to use a workflow_dispatch here because we're using the matrix branch value.
+      # Calling a workflow using the `uses` syntax does not give us access to any context so
+      # we must use hardcoded tags/branches/SHAs.  This also means we need to manually monitor
+      # for when the workflow completes.
+      - name: Call CI workflow for ${{ matrix.branch }} branch
+        id: trigger-step
+        run: |
+          gh workflow run ${{ matrix.workflow_name }} --ref ${{ matrix.branch }}
+        env:
+          GH_TOKEN: ${{ secrets.FISHTOWN_BOT_PAT }}
+
+      - name: Get workflow run ID
+        id: workflow-id
+        run: |
+           table=$(gh run list --workflow=${{ matrix.workflow_name }})
```

Comment:
If we update this to:
```
table=$(gh run list --workflow=${{ matrix.workflow_name }} --branch=${{ matrix.branch }})
```
Does that fix the race condition?

Reply:
It doesn't. And I'm honestly not totally sure why.  I tried it and when I add the branch flag it doesn't find the right run id and considers everything complete immediately.